git_machete/tests/functional/test_machete.py

```diff
@@ -26,6 +29,148 @@ def get_head_commit_hash() -> str:
+            self.update_issue()
+        elif 'pulls' in self.url:
+            self.update_pull()
+
+    def handle_post(self) -> None:
+        if 'issues' in self.url:
+            if self.url.endswith('issues'):
+                self.update_issue(new=True)
+        elif 'pulls' in self.url:
+            if self.url.endswith('pulls'):
+                self.update_pull(new=True)
+
+    def update_pull(self, new: bool = False) -> None:
+        pull: Dict[str, Any]
+        if new:
+            pull = {'number': self.get_next_free_number(GitAPIState.pulls),
+                    'user': {'login': 'github_user'},
+                    'html_url': 'www.github.com'}
+        else:
+            pull_no: str = self.find_number(self.url)
+            pull = self.get_pull(pull_no)
+        for key in json.loads(self.json_data).keys():
+            if key in ('base', 'head'):
+                pull[key] = {'ref': ""}
+                pull[key]['ref'] = json.loads(self.json_data)[key]
+            else:
+                pull[key] = json.loads(self.json_data)[key]
+        GitAPIState.pulls.append(pull)
+        self.status_code = 201
+        self.return_data = pull
+
+    def update_issue(self, new: bool = False) -> None:
+        issue: Dict[str, Any]
+        if new:
+            issue = {'number': self.get_next_free_number(GitAPIState.issues)}
+        else:
+            issue_no = self.find_number(self.url)
+            issue = self.get_issue(issue_no)
+        for key in json.loads(self.json_data).keys():
+            issue[key] = json.loads(self.json_data)[key]
+        GitAPIState.issues.append(issue)
+        self.status_code = 201
+        self.return_data = issue
+        return
+
+    @staticmethod
+    def get_issue(issue_no: str) -> Optional[Dict[str, Any]]:
+        for index, issue in enumerate(GitAPIState.issues):
+            if issue['number'] == issue_no:
+                return GitAPIState.issues.pop(index)
+        return None
+
+    @staticmethod
+    def get_pull(pull_no: str) -> Optional[Dict[str, Any]]:
+        for index, pull in enumerate(GitAPIState.pulls):
+            if pull['number'] == pull_no:
+                return GitAPIState.pulls.pop(index)
+        return None
+
+    @staticmethod
+    def find_number(url: str) -> str:
+        m = re.search(r'\d+', url)
+        return m.group()
+
+    @staticmethod
+    def get_next_free_number(entities: List[Dict[str, Any]]) -> str:
+        numbers = [int(item['number']) for item in entities]
+        return str(max(numbers) + 1)
+
+    @staticmethod
+    def set_initial_values() -> None:
+        GitAPIState.pulls = [
+            {'head': {'ref': 'bugfix/remove'}, 'user': {'login': 'github_user'}, 'base': {'ref': 'develop'},
+             'number': '15', 'html_url': 'www.github.com'}]
+        GitAPIState.user = {'login': 'github_user', 'type': 'User', 'company': 'VirtusLab'}
+        GitAPIState.issues = []
+        GitAPIState.remote_branches = []
+
+
+class ContextManager:
```

Comment:
It would be good to specify what kind of context it manages. Given its implementation and purpose I would suggest one of the followings:
- `MockContextManager` <- my preference
- `EmptyContextManager`
- `FakeContextManager`

Reply:
Right, that's what I suggested originally (albeit implicitly) by `...ContextManager` ;)